### ISSUE TITLE
bug: dust in non-cartesian geometries

### ISSUE THREAD

**Issue by neutrinoceros** [amrvac/amrvac#1]
I was having very odd results when using `physics/mod_dust.t` in polar (1.5D and 2D) geometries.
In `hd/mod_hd_phys.t` I found that piece of comment : 

```fortran
!> Ileyk : to do :
!>     - address the source term for the dust
subroutine hd_add_source_geom(qdt, ixI^L, ixO^L, wCT, w, x)
```
indeed, this routine still completely ignores dust. I'll see if I'm able to fix this.

**Comment by neutrinoceros** [comment]
I proposed a solution to this issue here
https://github.com/amrvac/amrvac/pull/3

**Comment by neutrinoceros** [comment]
solved ddeeb0c8631cb33e2d28823162409cb239eec5c8
